packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -176,16 +176,28 @@ export class VirtualNode extends VirtualNodeBase {
     props.listeners?.forEach(listener => this.addListener(listener));
     const accessLogging = props.accessLog?.bind(this);
     const serviceDiscovery = props.serviceDiscovery?.bind(this);
+    const backendDefaults = props.backendDefaults;
 
     const node = new CfnVirtualNode(this, 'Resource', {
       virtualNodeName: this.physicalName,
       meshName: this.mesh.meshName,
       spec: {
         backends: cdk.Lazy.any({ produce: () => this.backends }, { omitEmptyArray: true }),
         listeners: cdk.Lazy.any({ produce: () => this.listeners.map(listener => listener.listener) }, { omitEmptyArray: true }),
-        backendDefaults: props.backendDefaults !== undefined
-          ? {
-            clientPolicy: props.backendDefaults?.clientPolicy?.bind(this).clientPolicy,
+        backendDefaults: backendDefaults !== undefined ?
+          {
+            clientPolicy: backendDefaults.clientPolicy !== undefined ?
```

Comment:
We could reduce redundancy and have a single `ClientPolicy.bind()` function if we follow the implementation of `tlsCertificate`. They use inheritance [example](https://github.com/Seiya6329/aws-cdk/blob/master/packages/%40aws-cdk/aws-appmesh/lib/tls-certificate.ts#L103-L135).

Reply:
Would we not able to create `bind()` function for `ClientPolicy` since it is changed into interface?

I think `tlsValidationContext` is more comparable to `tlsCertificate` and I can update its implementation pattern to use inheritance. Please share your thought.

Reply:
Getting clarification from AJ, `bind()` method on `trust` and not on `ClientPolicy`.